### PR TITLE
Upgrade libbpf 1.2.2

### PR DESCRIPTION
**Pull request by sridhartigera**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf/tc/attach.go [hunk 1]
```diff
@@ -84,16 +84,20 @@ func (ap *AttachPoint) loadObject(ipVer int, file string) (*libbpf.Obj, error) {
 		// In case of global variables, libbpf creates an internal map <prog_name>.rodata
 		// The values are read only for the BPF programs, but can be set to a value from
 		// userspace before the program is loaded.
+		mapName := m.Name()
 		if m.IsMapInternal() {
+			if strings.HasPrefix(mapName, ".rodata") {
```
Comment (tomastigera) [inline felix/bpf/tc/attach.go:89]: What other maps are internal now? Would this help with movind strings from code to .rodata?

File: felix/bpf-gpl/tc6.c [hunk 2]
```diff
@@ -24,7 +24,7 @@
 const volatile struct cali_tc_globals __globals;
 
 
-SEC("classifier/tc/prologue")
+SEC("classifier")
```
Comment (tomastigera) [inline felix/bpf-gpl/tc6.c:27]: Why did we need to place everything into a single section now?

Comment (sridhartigera) [reply felix/bpf-gpl/tc6.c]: The way the section name is parsed has changed. In earlier versions, ```classifier/x/y``` was allowed but in the latest it is just ```classifier```.

Comment (sridhartigera) [reply felix/bpf-gpl/tc6.c]: I have also changed the section name from ```classifier``` to ```tc``` as ```classifier``` is deprecated from v1.0.0

File: felix/bpf/tc/attach.go (hunk 1 above)
Comment (sridhartigera) [reply felix/bpf/tc/attach.go:89]: If you remember, we used to get unrecognized section .rodata warning. Now they have defined these as separate maps (BPF_MAP_TYPE_ARRAY). These maps along with the global data is marked as RODATA only difference being global data maps are named as <map_name>.<rodata> whereas these maps are named .rodata.1.1. Hence we need to skip these maps. I am sure about string handling. Have to check how that works.

**Pull request merged**